plumpy/processes.py

```diff
@@ -481,17 +467,24 @@ def _process_scope(self):
         meaning that globally someone can ask for Process.current() to get the last process
         that is on the call stack.
         """
-        _process_stack().append(self)
+        stack_copy = PROCESS_STACK.get().copy()
+        stack_copy.append(self)
+        PROCESS_STACK.set(stack_copy)
```

Comment:
Can you add a description on why we cannot do the append operation on the global variable? Same below for the popping